frontend/hooks/useNotificationNote.js

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useRef } from 'react';
+import * as Notifications from 'expo-notifications';
+
+Notifications.setNotificationHandler({
+    handleNotification: async () => ({
+        shouldShowAlert: true,
+        shouldPlaySound: true,
+        shouldSetBadge: true,
+    }),
+});
+
+export default function useNotificationNote(callback) {
+    const responseListener = useRef();
+
+    useEffect(() => {
+        responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {
```

Comment:
we add here a lsitener to listen to the event of getting notification
In case we got a notification, we extract the note from notification (response.notification.request.content.data) and run callback with this note